### PR TITLE
Refactored SelectMetadataField.jsx to functional component

### PR DESCRIPTION
**Pull request by lorstenoplo**
Ref: #4460 

I have converted the  src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx to a functional component and removed unused code.

### CODE REVIEW

**Comment by dhruvgupta0503** [comment]
Hi, Can I work on this issue

**Comment by stevepiercy** [comment]
@dhruvgupta0503 no, it is a pull request. I suggest you learn the difference between GitHub issues and pull requests by reading its documentation.

Also please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do).

File: .gitpod.yml
Comment (stevepiercy) [inline .gitpod.yml:1]: This file should not be part of this pull request. If you want to add support for Gitpod, please open a new and separate issue for discussion. Until then, please remove this file from the pull request or add it to your user's `.gitignore`.

File: packages/volto/news/5570.internal [hunk 1]
```diff
@@ -0,0 +1 @@
+Refactored src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx to functional component. @lorstenoplo
```
Comment (stevepiercy) [inline packages/volto/news/5570.internal:1]:
```suggestion
Refactored the `SelectWidget` component to a functional component. @lorstenoplo
```

File: packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx [hunk 2]
```diff
@@ -30,187 +30,158 @@ const messages = defineMessages({
...
+  //   return null;
+  // };
+
+  // const handleChange = (selectedOption) => {
+  //   setSelectedOption(selectedOption);
+  //   props.onChange(props.id, {
+  //     value: selectedOption.value,
+  //     title: selectedOption.label,
+  //   });
+  // };
```
Comment (stevepiercy) [inline packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx:84]: Commenting code as you have done twice above is a sloppy development practice and makes reviewing `git diff` difficult. Delete the commented code.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Comment by stevepiercy** [comment]
@lorstenoplo please see Item 2 in https://6.docs.plone.org/contributing/first-time.html#create-a-pull-request-from-your-fork and edit your pull request description.

File: packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx (hunk 2 above)
Comment (dhruvgupta0503) [reply packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx:84]: ok sir , I will delete the commented code

File: .gitpod.yml
Comment (lorstenoplo) [reply .gitpod.yml:1]: I removed it

File: packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx (hunk 2 above)
Comment (lorstenoplo) [reply packages/volto/src/components/manage/Blocks/Search/widgets/SelectMetadataField.jsx]: Deleted the commented code

**Comment by stevepiercy** [comment]
Final step, @lorstenoplo please see Item 2 in https://6.docs.plone.org/contributing/first-time.html#create-a-pull-request-from-your-fork and edit your pull request description. Then it can be reviewed.

**Review by stevepiercy — APPROVED** [review]
I triaged this PR for review by a core developer. It's ready.

**Comment by lorstenoplo** [comment]
Why is the pr not being merged?

**Comment by stevepiercy** [comment]
@plone/volto-team this PR needs review.

**Review by tiberiuichim — APPROVED** [review]
